src/physics/utils/pio_reader.F90

```diff
@@ -1800,22 +1281,33 @@ subroutine get_netcdf_var_char_0d(this, varname, var, errmsg, errcode)
       use pio,        only: PIO_BCAST_ERROR
       use pio_types,  only: PIO_CHAR
 
-      class(pio_reader_t),       intent(in)  :: this
-      character(len=*),          intent(in)  :: varname
-      character(len=:), pointer, intent(out) :: var     !Character variable that file data will be read to.
-      integer,                   intent(out) :: errcode !Error code
-      character(len=*),          intent(out) :: errmsg  !Error message
+      class(pio_reader_t),           intent(in)  :: this
+      character(len=*),              intent(in)  :: varname
+      character(len=:), allocatable, intent(out) :: var     !Character variable that file data will be read to.
+      integer,                       intent(out) :: errcode !Error code
+      character(len=*),              intent(out) :: errmsg  !Error message
+
+      !Optional arguments for reading a subset of the variable
+      integer, optional,             intent(in)  :: start(:) !Start indices for each dimension
+      integer, optional,             intent(in)  :: count(:) !Number of elements to read for each dimension
 
       !Local variables:
       type(file_desc_t)    :: pio_file_handle !File handle type used by PIO
       character(len=cl)    :: file_path       !Path to NetCDF file
       integer              :: err_handling    !PIO error handling code
       integer              :: var_id          !NetCDF variable ID
       integer              :: nc_type         !NetCDF variable type
-      integer              :: ndims           !Number of variable dimensions on NetCDF file
-      integer, allocatable :: dim_ids(:)      !Variable dimension IDs
       integer, allocatable :: dim_sizes(:)    !Variable dimension sizes
-      integer              :: i !loop control variable
+
+      !NOTE:  NetCDF supports both character arrays and string-type
+      !data depending on the NetCDF version, so the dimensions
+      !might be one larger than the actual array size if it
+      !includes the character length as a dimension as well.
+      !Ideally the actual type would be checked and handled
+      !differently, but for now just assume a character array
+      !and check for ndims = rank+1
```

Comment:
is this comment still accurate? Aren't we checking the type?

Reply:
We are only making sure that the type is actually a "CHAR" type variable.  Ideally we would check for either "CHAR" or "STRING", and then change the  behavior/variable allocation accordingly, but given that I am not sure how well supported the "STRING" type is in Fortran at the moment, I decided to put that functionality on hold for now.

Of course if you would like me to change the comment or anything please let me know!

Reply:
maybe this would be a bit clearer?
```suggestion
      !Ideally the actual type would be checked and handled
      !differently, but for now just confirm it's a character array
      !and check for ndims = rank+1
```